packages/angular/ssr/src/routes/ng-routes.ts

```diff
@@ -27,7 +28,29 @@ import { Console } from '../console';
 import { AngularAppManifest, getAngularAppManifest } from '../manifest';
 import { AngularBootstrap, isNgModule } from '../utils/ng';
 import { joinUrlParts } from '../utils/url';
-import { RouteTree } from './route-tree';
+import { RenderMode, SERVER_ROUTES_CONFIG, SSGFallback, ServerRoute } from './route-config';
+import { RouteTree, RouteTreeNodeMetadata } from './route-tree';
+
+/**
+ * Regular expression to match segments preceded by a colon in a string.
+ */
+const URL_PARAMETER_REGEXP = /:([^/]+)(?=\/|$)/g;
+
+/**
+ * An set of HTTP status codes that are considered valid for redirect responses.
+ */
+const VALID_REDIRECT_RESPONSE_CODES = new Set([301, 302, 303, 307, 308]);
+
+/**
+ * Additional metadata for a server configuration route tree.
+ */
+type ServerConfigRouteTreeAdditionaMetadata = Record<string, unknown> & Partial<ServerRoute>;
```

Comment:
Question: What data are we actually putting into this untyped "additional metadata" value? Is there a stronger abstraction we could define for this or potentially wrap the existing metadata types rather than extending them like this?

Reply:
Removing `Record<string, unknown>`